File: TestItNow/TINProducts.m
//
//  TINProducts.m
//  TestItNow
//
//  Created by Michael Soares on 7/9/15.
//  Copyright (c) 2015 Postmates. All rights reserved.
//

#import "TINProducts.h"

@implementation TINProducts

+ (NSArray *)products
{
    static NSArray *s_products;
    static dispatch_once_t onceToken;
    dispatch_once(&onceToken, ^{
        // we're not going to worry about error handling while reading in this file
        // we can safely assume for the purpose of this project that the JSON is sane
        NSString *jsonFilePath = [[NSBundle mainBundle] pathForResource:@"products" ofType:@"json"];
        NSData *jsonData = [NSData dataWithContentsOfFile:jsonFilePath];
        NSError *error;
        s_products = [NSJSONSerialization JSONObjectWithData:jsonData
                                                     options:NSJSONReadingMutableContainers
                                                       error:&error];
    });
    
    return s_products;
}

@end

File: TestItNow/TINProducts.swift
//
//  TINProducts.swift
//  TestItNow
//
//  Created by Kevin Ballard on 7/11/16.
//  Copyright © 2016 Postmates. All rights reserved.
//

import Foundation

class TINProducts: NSObject {
    static let products: [[AnyHashable: Any]] = {
        // we're not going to worry about error handling while reading in this file
        // we can safely assume for the purpose of this project that the JSON is sane
        let jsonFilePath = Bundle.main.path(forResource: "products", ofType: "json")!
        let jsonData = try! Data(contentsOf: URL(fileURLWithPath: jsonFilePath))
        return try! JSONSerialization.jsonObject(with: jsonData, options: .mutableContainers) as! [[AnyHashable: Any]]
    }()
}
